project-koku/koku
# Cost 1329 final azure enabled tag key migration

## Jira Ticket

[COST-1329](https://issues.redhat.com/browse/COST-1329)

## Dependency

#3717  must be released to staging and prod first.

## Description

This is a migration that will remove the `id` column and unique constraint on `key` and create a primary key constraint on `key`

## Testing

1. Checkout branch
2. Run migrations `make run-migrations`
3. `psql postgresql://postgres@localhost:15432/postgres`
    1. `\d template0.reporting_azureenabledtagkeys` 
    2. Verify that there is no `id` column and the primary key constraint is on the `key` column
    3. \d template0.reporting_gcpenabledtagkeys` 
    4. Verify that the `id` column exists
    5. Exit `psql`
4. Run reverse migration `make run-migrations applabel=reporting migration=0256
6. `psql postgresql://postgres@localhost:15432/postgres`
    1. `\d template0.reporting_azureenabledtagkeys` 
    2. Verify that the `id` column exists and is the primary key and the `key` column only has a unique constraint
    3. \d template0.reporting_gcpenabledtagkeys` 
    4. Verify that the `id` column exists
    5. Exit `psql`

## Notes
